compiler/src/dotty/tools/dotc/core/NameKinds.scala

```diff
@@ -61,8 +61,7 @@ object NameKinds {
     def infoString: String
   }
 
-  /** The kind of SimpleNames */
-  val SimpleNameKind = new NameKind(UTF8) { self =>
+  object SimpleNameKind extends NameKind(UTF8) { self =>
```

Comment:
Ah, I see, indeed that's different. If we had enums we could iterate over the namekinds to register them maybe :).